src/MacroTools/Mechanics/StructurePacking.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using WCSharp.Events;
+using static War3Api.Common;
+
+namespace MacroTools.Mechanics
+{
+  /// <summary>
+  /// System to allow structures to convert themselves into units. Those units can then use an ability to rebuild the building.
+  /// </summary>
+  public class StructurePacking
+  {
+    /// <summary>
+    /// Object that contains information and methods for packable structures
+    /// </summary>
+    public class PackableStructure
+    {
+      private static readonly Dictionary<int, PackableStructure> PackableStructureById = new();
+      private int _packedUnitId;
+      private int _buildAbility;
+      private string? _structureModel;
+
+      /// <summary>
+      /// Method for creating new packable structure entries
+      /// </summary>
+      /// <param name="structureId"></param>
+      /// <param name="packedUnitId"></param>
+      /// <param name="buildAbility"></param>
+      /// <param name="structureModel"></param>
+      public static void Create(int structureId, int packedUnitId, int buildAbility, string? structureModel)
```

Comment:
Could we do this for consistency with some of the other systems?
```suggestion
      public static void Register(int structureId, int packedUnitId, int buildAbility, string? structureModel)
```